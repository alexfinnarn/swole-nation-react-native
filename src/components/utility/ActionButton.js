import React from 'react';
import {Text, TouchableOpacity, View} from 'react-native';
import {home} from "../Styles";

export default function ActionButton({text, action, label, styles, disabled = false}) {
  const bgColor = disabled ? '#bababa' : '#21897E';

  return (
    <View style={[{flex: 1, flexDirection: 'column'}, styles]}>
      <TouchableOpacity
        disabled={disabled}
        accessible={true}
        accessibilityLabel={label}
        activeOpacity={0.6}
        style={[{backgroundColor: bgColor}, home.actionButton]}
        onPress={action}>
        <View style={{flex: 1, justifyContent: 'center'}}>
          <Text style={home.actionButtonText}>{text}</Text>
        </View>
      </TouchableOpacity>
    </View>
  );
}
