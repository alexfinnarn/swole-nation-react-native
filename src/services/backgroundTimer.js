import BackgroundTimer from 'react-native-background-timer';
import * as Speech from 'expo-speech';

const timer = (function() {
  this.timeDurations = [
    '90 seconds',
    '3 minutes'
  ];
  this.counter = 0;

  return {
    start: () => {
      BackgroundTimer.runBackgroundTimer(() => {
        if (typeof this.timeDurations[this.counter] !== 'undefined') {
          // Speech.speak('do the thing, mayne. Fool! ' + seconds + ' have passed', {pitch: 2});
          console.log(`Time for your next set. ${this.timeDurations[this.counter]} has passed.`);
        }
        this.counter++;
      }, (5000));
    },
    stop: () => {
      BackgroundTimer.stopBackgroundTimer();
      this.counter = 0;
    }
  }
}());

export default timer;
