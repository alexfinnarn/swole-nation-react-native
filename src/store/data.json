{
  "workouts": [
    {
      "name": "Stronglifts A",
      "key": "DQkECwYLCQQ",
      "description": "Squats, Bench Press, Barbell Row",
      "exercises": [
        "Squats Warmup"
      ]
    }
  ],
  "exercises": [
    {
      "name": "Squats Warmup",
      "key": "BwEBDwoFBgQ",
      "image": "../../assets/squats.gif",
      "instructions": "Do with a barbell",
      "sets": [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 3,
          "weight": 185.0
        },
        {
          "reps": 2,
          "weight": 205.0
        }
      ]
    }
  ],
  "sessions": [
    {
      "name": "new ...date",
      "key": "BwgFDQ8CCg8",
      "workoutKeys": [],
      "exercises": []
    }
  ],
  "tempExerciseList": [],
  "theThing": "",
  "activeWorkoutIndex": 0,
  "activeSessionIndex": 0,
  "activeExerciseIndex": 0
}
