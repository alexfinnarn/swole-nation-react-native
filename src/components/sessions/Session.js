import React, {useState, useEffect} from 'react';
import {Text, View, Button, Image} from "react-native";
import BackgroundTimer from 'react-native-background-timer';
import * as Speech from 'expo-speech';
import {styles} from "../Styles";
import Table from "../utility/Table";
import ActionButton from "../utility/ActionButton";

export default function Session({session, navigation, finishSession}) {

  const [set, updateSet] = useState(0);
  const [exercise, updateExercise] = useState(0);

  const [seconds, setSeconds] = useState(0);
  const [sessionDuration, setSessionDuration] = useState(0);
  const [isActive, setIsActive] = useState(false);
  const [weightPlateString, setWeightPlateString] = useState('Each Side: ');
  const weightPlates = [45, 25, 10, 5, 2.5];

  // Create timer used for workout.
  useEffect(() => {
    let interval = null;
    if (isActive) {
      interval = setInterval(() => {
        setSeconds(seconds => seconds + 1);
      }, 1000);
    } else if (!isActive && seconds !== 0) {
      clearInterval(interval);
    }
    return () => clearInterval(interval);
  }, [isActive, seconds]);

  // Start timer at the beginning of workout and set initial weight plate string.
  useEffect(() => {
    setTimeout(() => {
      toggle();
      calculateWeightPlates((session.exercises[exercise].sets[set].weight - 45.0) / 2);
    }, 200);
  }, []);

  function toggle() {
    setIsActive(!isActive);
  }

  function resetTimer() {
    setSessionDuration(sessionDuration + seconds);
    setSeconds(0);
  }

  function calculateWeightPlates(weight, platesString = '', counter = 0) {
    if (weight <= 0 && counter >= weightPlates.length) {
      // Chop off first three characters hyphen for display.
      setWeightPlateString(platesString.slice(2));
    } else {

      // The 45 and 25 pound weight plates can be included multiple times.
      // The 25 only twice, but the 45 can be added many more times.
      // To display them, we add a multiplier to the display to tell the user how many of the plates
      // to put on the barbell.
      let multiplier = 1;
      if ((weight / weightPlates[counter]) >= 2 && [45, 10].includes(weightPlates[counter])) {
        multiplier = Math.floor(weight / weightPlates[counter]);
      }

      if ((weight - weightPlates[counter]) >= 0) {
        weight = weight - (weightPlates[counter] * multiplier);
        platesString = `${platesString} - ${multiplier.toString()}x${weightPlates[counter].toString()}`;
      }

      calculateWeightPlates(weight, platesString, counter + 1);
    }
  }

  function handleSets(forward, completed = false) {
    if (completed) {
      session.exercises[exercise].sets[set].completed = true;
    }

    // If forwards and at the end of the sets, move to the next exercise.
    if (forward && set === session.exercises[exercise].sets.length - 1) {
      updateExercise(exercise + 1);
      updateSet(0);
      calculateWeightPlates((session.exercises[exercise + 1].sets[0].weight - 45.0) / 2);
      resetTimer();
      return;
    }

    // If backwards and at the beginning of sets, then move back an exercise.
    if (!forward && set === 0) {
      updateExercise(exercise - 1);
      updateSet(session.exercises[exercise - 1].sets.length - 1);
      calculateWeightPlates((session.exercises[exercise - 1].sets[session.exercises[exercise - 1].sets.length - 1].weight - 45.0) / 2);
      return;
    }

    // BackgroundTimer.stopBackgroundTimer();
    // BackgroundTimer.runBackgroundTimer(() => {
    //   Speech.speak('do the thing, mayne. Fool!');
    // }, (1000 * 10));

    if (forward) {
      resetTimer();
      updateSet(set + 1);
      calculateWeightPlates((session.exercises[exercise].sets[set].weight - 45.0) / 2);
    } else {
      updateSet(set - 1);
      calculateWeightPlates((session.exercises[exercise].sets[set].weight - 45.0) / 2);
    }
  }

  function onLastExercise() {
    return session.exercises.length - 1 === exercise && session.exercises[exercise].sets.length - 1 === set;
  }

  function finishWorkout() {
    session.duration = seconds + sessionDuration;
    finishSession(session);
    navigation.navigate('SessionsList');
  }

  function ActionButtons() {
    let last, current, next = '';
    if (!session.exercises[exercise - 1] && set === 0) {
      last = <ActionButton key="previous" disabled={true} text="Previous" action={() => {}}/>;
    } else {
      last = <ActionButton key="previous" text="Previous" action={() => handleSets(false)}/>;
    }

    if (onLastExercise()) {
      current = <ActionButton styles={{paddingLeft: 5, paddingRight: 5}} key="current" text="Finish" action={() => finishWorkout()}/>;
      next = <ActionButton key="next" disabled={true} text="Skip" action={() => {}}/>;
    } else {
      current = <ActionButton styles={{paddingLeft: 5, paddingRight: 5}} key="current" text="Complete" action={() => handleSets(true, true)}/>;
      next = <ActionButton key="next" text="Skip" action={() => handleSets(true)}/>;
    }

    return ([last, current, next]);
  }

  function StyleTile({color, bgColor, text, flex}) {
    return (
      <View style={{flex: flex, backgroundColor: bgColor}}>
        <Text style={{flex: 1, color: color, alignSelf: 'center'}}>
          {text}
        </Text>
      </View>
    );
  }

  return (
    <View style={[styles.container, {flex: 1, flexDirection: 'column', justifyContent: 'space-between'}]}>
      <View style={{flex: 5, padding: 10, flexDirection: 'row'}}>
        <View style={{flex: 3, justifyContent: 'center', alignItems: 'center'}}>
          <Image resizeMode="contain" style={{flex: 1}} source={session.exercises[exercise].image}/>
        </View>
        <View style={{flex: 1, flexDirection: 'column'}}>
          <StyleTile color={'#ffffff'} bgColor={'#69D1C5'} flex={1}
                     text={session.exercises[exercise - 1] ? session.exercises[exercise - 1].name : ''}/>
          <StyleTile color={'#ffffff'} bgColor={'#3BA99C'} flex={2}
                     text={session.exercises[exercise].name}/>
          <StyleTile color={'#ffffff'} bgColor={'#21897E'} flex={1}
                     text={session.exercises[exercise + 1] ? session.exercises[exercise + 1].name : ''}/>
        </View>
      </View>
      <View style={{flex: 1, padding: 5, flexDirection: 'row', justifyContent: 'space-between', alignItems: 'flex-end'}}>
        <ActionButtons/>
      </View>
      <View style={{flex: 4, padding: 10, flexDirection: 'column', justifyContent: 'space-around'}}>
        <Table
          columnFlex={[1, 1, 1, 1]}
          headers={['Completed', 'Sets', 'Reps', 'Weight']}
          rowData={[
            // session.exercises[exercise].sets[set].completed.toString(),
            'foo',
            `${set + 1}/${session.exercises[exercise].sets.length}`,
            session.exercises[exercise].sets[set].reps,
            `${session.exercises[exercise].sets[set].weight} lbs`
          ]}
        />
        <Table
          columnFlex={[1, 3, 1]}
          headers={['Set', 'Plates', 'Session']}
          rowData={[seconds, weightPlateString, seconds + sessionDuration]}
        />
      </View>
      <View style={{flex: 1, padding: 10, flexDirection: 'row', justifyContent: 'space-evenly', alignItems: 'center'}}>
        <ActionButton styles={{paddingRight: 5}} text="Quit" action={() => {navigation.navigate('Home');}}/>
        <ActionButton text="Pause" action={() => toggle()}/>
      </View>
    </View>
  );
}
