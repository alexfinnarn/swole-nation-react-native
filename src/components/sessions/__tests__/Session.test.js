import React from 'react';
import Session from '../Session';
import {render, fireEvent, act} from 'react-native-testing-library';
import {Image} from 'react-native';
import data from '../../../store/data';
import ActionButton from "../../utility/ActionButton";
import backgroundTimer from '../../../services/backgroundTimer';

const timerStop = jest.spyOn(backgroundTimer, 'stop');
const timerStart = jest.spyOn(backgroundTimer, 'start');

let renderer, instance = {};
const strongliftsA = 'DQkECwYLCQQ';

function getSession() {
  let session = {
    key: 'rQksCwYMCQF',
    duration: 0,
    name: new Date(Date.now()).toLocaleString('en-US'),
    workoutKeys: [data.workouts[strongliftsA].key],
    exercises: data.workouts[strongliftsA].exercises.map((name) => {
      const foundKey = Object.keys(data.exercises).find(key => data.exercises[key].name === name);
      return data.exercises[foundKey];
    })
  };

  // Add completed property. Not added to workout because it only makes sense to record in a session.
  session.exercises.map((exercise) => {
    exercise.sets = exercise.sets.map((set) => {
      set.completed = false;
      return set;
    });
    return exercise;
  });

  return session;
}

const handle = {
  setActiveSessionKey: jest.fn(),
};
const navigation = {
  navigate: jest.fn()
};

beforeEach(() => {
  renderer = render(<Session session={getSession()} navigation={navigation} handle={handle}/>);
  instance = renderer.getByTestId('session-root');
}, 0);

jest.useFakeTimers();

describe('<Session />', () => {
  it('Renders image and exercise list in the action display area and changes them when sets progress forward', () => {
    let image = instance.findAllByType(Image);
    expect(image.length).toBe(1);
    expect(image[0].props.testID).toBe('squats-warmup-image');
    expect(image[0].props.testID).not.toBe('bench-press-image');

    expect(renderer.queryByText('Squats Warmup')).not.toBeNull();
    expect(renderer.queryByText('Bench Press')).toBeNull();

    [...Array(15)].forEach(() => fireEvent(renderer.getByText('Complete'), 'press'));

    image = instance.findAllByType(Image);
    expect(image.length).toBe(1);
    expect(image[0].props.testID).toBe('bench-press-image');
    expect(image[0].props.testID).not.toBe('squats-warmup-image');

    expect(renderer.queryByText('Bench Press')).not.toBeNull();
    expect(renderer.queryByText('Squats Warmup')).toBeNull();
  });

  it('moves set list back and forward when progress buttons are pressed and complete acts different from skip', () => {
    let actionButtons = instance.findAllByType(ActionButton);

    expect(renderer.queryByTestId('sets-reps-table-sets').props.children).toBe('1/6');
    expect(renderer.queryByTestId('sets-reps-table-sets').props.children).not.toBe('2/6');
    expect(renderer.queryByTestId('sets-reps-table-completed').props.children).toBe('false');
    expect(renderer.queryByTestId('sets-reps-table-reps').props.children).toBe('5');
    expect(renderer.queryByTestId('sets-reps-table-weight').props.children).toBe('45 lbs');

    // Previous button is disabled on the first set.
    expect(actionButtons[0].props.disabled).toBeDefined();

    fireEvent(renderer.getByText('Complete'), 'press');

    expect(renderer.queryByTestId('sets-reps-table-sets').props.children).toBe('2/6');
    expect(renderer.queryByTestId('sets-reps-table-sets').props.children).not.toBe('1/6');
    expect(renderer.queryByTestId('sets-reps-table-completed').props.children).toBe('false');

    // Previous button is no longer disabled.
    actionButtons = instance.findAllByType(ActionButton);
    expect(actionButtons[0].props.disabled).toBeUndefined();

    fireEvent(renderer.getByText('Previous'), 'press');

    expect(renderer.queryByTestId('sets-reps-table-sets').props.children).toBe('1/6');
    expect(renderer.queryByTestId('sets-reps-table-sets').props.children).not.toBe('2/6');
    expect(renderer.queryByTestId('sets-reps-table-completed').props.children).toBe('true');

    actionButtons = instance.findAllByType(ActionButton);
    expect(actionButtons[0].props.disabled).toBeDefined();

    [...Array(2)].forEach(() => fireEvent(renderer.getByText('Skip'), 'press'));

    expect(renderer.queryByTestId('sets-reps-table-sets').props.children).toBe('3/6');
    expect(renderer.queryByTestId('sets-reps-table-sets').props.children).not.toBe('2/6');
    expect(renderer.queryByTestId('sets-reps-table-completed').props.children).toBe('false');
    expect(renderer.queryByTestId('sets-reps-table-weight').props.children).toBe('95 lbs');

    fireEvent(renderer.getByText('Previous'), 'press');

    expect(renderer.queryByTestId('sets-reps-table-sets').props.children).toBe('2/6');
    expect(renderer.queryByTestId('sets-reps-table-sets').props.children).not.toBe('3/6');
    expect(renderer.queryByTestId('sets-reps-table-completed').props.children).toBe('false');
    expect(renderer.queryByTestId('sets-reps-table-weight').props.children).toBe('45 lbs');

    [...Array(3)].forEach(() => fireEvent(renderer.getByText('Skip'), 'press'));
    expect(renderer.queryByTestId('sets-reps-table-reps').props.children).toBe('3');
  });

  it('Weight plates string displays correctly as sets move forward', () => {
    // The weight plates string begins with a space to separate it from the rest of the plates.
    expect(renderer.queryByTestId('time-plates-table-plates').props.children).toBe(' None');

    [...Array(2)].forEach(() => fireEvent(renderer.getByText('Skip'), 'press'));
    expect(renderer.queryByTestId('time-plates-table-plates').props.children).toBe(' 1x25');

    fireEvent(renderer.getByText('Skip'), 'press');
    expect(renderer.queryByTestId('time-plates-table-plates').props.children).toBe(' 1x45');

    fireEvent(renderer.getByText('Complete'), 'press');
    expect(renderer.queryByTestId('time-plates-table-plates').props.children).toBe(' 1x45 - 1x25');

    fireEvent(renderer.getByText('Skip'), 'press');
    expect(renderer.queryByTestId('time-plates-table-plates').props.children).toBe(' 1x45 - 1x25 - 1x10');

    fireEvent(renderer.getByText('Complete'), 'press');
    expect(renderer.queryByTestId('time-plates-table-plates').props.children).toBe(' 2x45 - 1x2.5');
  });

  it('Timer progresses and set timer resets when new set and exercise proceed', () => {


    expect(renderer.queryByTestId('time-plates-table-set').props.children).toBe(0);
    expect(renderer.queryByTestId('time-plates-table-session').props.children).toBe(0);

    jest.advanceTimersByTime(4000);

    expect(renderer.queryByTestId('time-plates-table-set').props.children).toBe(3);
    expect(renderer.queryByTestId('time-plates-table-session').props.children).toBe(3);

    fireEvent(renderer.getByText('Complete'), 'press');
    expect(renderer.queryByTestId('time-plates-table-set').props.children).toBe(0);
    expect(renderer.queryByTestId('time-plates-table-session').props.children).toBe(3);

    jest.advanceTimersByTime(2000);

    [...Array(4)].forEach(() => fireEvent(renderer.getByText('Complete'), 'press'));
    expect(renderer.queryByTestId('time-plates-table-set').props.children).toBe(0);
    expect(renderer.queryByTestId('time-plates-table-session').props.children).not.toBe(0);

    // Moves on to next exercise.
    fireEvent(renderer.getByText('Complete'), 'press');
    expect(renderer.queryByTestId('time-plates-table-set').props.children).toBe(0);
    expect(renderer.queryByTestId('time-plates-table-session').props.children).not.toBe(0);
  });

  it('Pause and quit buttons work', () => {
    fireEvent(renderer.getByText('Quit'), 'press');
    // expect(timerStop).toHaveBeenCalledTimes(1);
    expect(navigation.navigate).toHaveBeenCalledTimes(1);
    expect(navigation.navigate).toHaveBeenCalledWith('Home');

    jest.useFakeTimers();
    expect(renderer.queryByTestId('time-plates-table-set').props.children).toBe(0);
    expect(renderer.queryByTestId('time-plates-table-session').props.children).toBe(0);

    jest.advanceTimersByTime(4000);
    expect(renderer.queryByTestId('time-plates-table-session').props.children).toBe(3);

    fireEvent(renderer.getByText('Pause'), 'press');
    jest.advanceTimersByTime(4000);
    expect(renderer.queryByTestId('time-plates-table-session').props.children).toBe(3);

    fireEvent(renderer.getByText('Pause'), 'press');
    jest.advanceTimersByTime(4000);
    expect(renderer.queryByTestId('time-plates-table-session').props.children).toBe(7);
    expect(timerStop).toHaveBeenCalledTimes(43);
  });
});
