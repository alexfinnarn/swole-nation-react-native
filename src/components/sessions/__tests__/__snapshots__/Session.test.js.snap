// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Session /> Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "flexDirection": "column",
        "paddingTop": 5,
      },
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      },
    ]
  }
  testID="session-root"
>
  <View
    style={
      Object {
        "flex": 5,
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "flex": 1,
          }
        }
        testID="squats-warmup-image"
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#69D1C5",
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#ffffff",
              "flex": 1,
            }
          }
          testID=""
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#3BA99C",
            "flex": 2,
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#ffffff",
              "flex": 1,
            }
          }
          testID="current-exercise"
        >
          Squats Warmup
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#21897E",
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#ffffff",
              "flex": 1,
            }
          }
          testID=""
        >
          Squats
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#bababa",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontSize": 20,
              }
            }
          >
            Previous
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
          },
          Object {
            "paddingLeft": 5,
            "paddingRight": 5,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#21897E",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontSize": 20,
              }
            }
          >
            Complete
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#21897E",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontSize": 20,
              }
            }
          >
            Skip
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 4,
        "flexDirection": "column",
        "justifyContent": "space-around",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
            },
            Object {
              "alignSelf": "stretch",
              "borderColor": "#d6d7da",
              "borderWidth": 1,
              "flex": 1,
              "paddingLeft": 7,
            },
          ]
        }
      >
        Completed
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
            },
            Object {
              "alignSelf": "stretch",
              "borderColor": "#d6d7da",
              "borderWidth": 1,
              "flex": 1,
              "paddingLeft": 7,
            },
          ]
        }
      >
        Sets
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
            },
            Object {
              "alignSelf": "stretch",
              "borderColor": "#d6d7da",
              "borderWidth": 1,
              "flex": 1,
              "paddingLeft": 7,
            },
          ]
        }
      >
        Reps
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
            },
            Object {
              "alignSelf": "stretch",
              "borderColor": "#d6d7da",
              "borderWidth": 1,
              "flex": 1,
              "paddingLeft": 7,
            },
          ]
        }
      >
        Weight
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "flexDirection": "row",
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "borderColor": "#d6d7da",
            "borderWidth": 1,
            "flex": 1,
            "paddingLeft": 7,
          }
        }
        testID="sets-reps-table-completed"
      >
        false
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "borderColor": "#d6d7da",
            "borderWidth": 1,
            "flex": 1,
            "paddingLeft": 7,
          }
        }
        testID="sets-reps-table-sets"
      >
        1/6
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "borderColor": "#d6d7da",
            "borderWidth": 1,
            "flex": 1,
            "paddingLeft": 7,
          }
        }
        testID="sets-reps-table-reps"
      >
        5
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "borderColor": "#d6d7da",
            "borderWidth": 1,
            "flex": 1,
            "paddingLeft": 7,
          }
        }
        testID="sets-reps-table-weight"
      >
        45 lbs
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
            },
            Object {
              "alignSelf": "stretch",
              "borderColor": "#d6d7da",
              "borderWidth": 1,
              "flex": 1,
              "paddingLeft": 7,
            },
          ]
        }
      >
        Set
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
            },
            Object {
              "alignSelf": "stretch",
              "borderColor": "#d6d7da",
              "borderWidth": 1,
              "flex": 3,
              "paddingLeft": 7,
            },
          ]
        }
      >
        Plates
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
            },
            Object {
              "alignSelf": "stretch",
              "borderColor": "#d6d7da",
              "borderWidth": 1,
              "flex": 1,
              "paddingLeft": 7,
            },
          ]
        }
      >
        Session
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "flexDirection": "row",
          "marginBottom": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "borderColor": "#d6d7da",
            "borderWidth": 1,
            "flex": 1,
            "paddingLeft": 7,
          }
        }
        testID="time-plates-table-set"
      >
        0
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "borderColor": "#d6d7da",
            "borderWidth": 1,
            "flex": 3,
            "paddingLeft": 7,
          }
        }
        testID="time-plates-table-plates"
      >
         None
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "borderColor": "#d6d7da",
            "borderWidth": 1,
            "flex": 1,
            "paddingLeft": 7,
          }
        }
        testID="time-plates-table-session"
      >
        0
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
          },
          Object {
            "paddingRight": 5,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#21897E",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontSize": 20,
              }
            }
          >
            Start
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#21897E",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontSize": 20,
              }
            }
          >
            Quit
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
